src/EventStore.Core/Services/Transport/Enumerators/Enumerator.StreamSubscription.cs

```diff
@@ -371,29 +379,6 @@ public class StreamSubscription<TStreamId> : StreamSubscription {
 							return;
 					}
 				}
-
-				void ConsumerTooSlow(ResolvedEvent evt) {
-					if (Interlocked.Exchange(ref liveMessagesCancelled, 1) != 0)
-						return;
-
-					var state = caughtUpSource.Task.Status == TaskStatus.RanToCompletion ? "live" : "transitioning to live";
-					var msg = $"Consumer too slow to handle event while {state}. Client resubscription required.";
-					Log.Information(
-						"Live subscription {subscriptionId} to {streamName} timed out at {streamRevision}. {msg}. unsubscribing...",
-						_subscriptionId, _streamName,
-						StreamRevision.FromInt64(evt.OriginalEventNumber), msg);
-
-					Unsubscribe();
-
-					liveEvents.Writer.Complete();
-
-					Fail(new ReadResponseException.Timeout(msg));
-				}
-
-				void Fail(Exception exception) {
-					this.Fail(exception);
-					caughtUpSource.TrySetException(exception);
-				}
 			}
 
 			private ValueTask ConfirmSubscription() => Interlocked.CompareExchange(ref _subscriptionStarted, 1, 0) != 0
```

Comment:
i think we might be able to simplify this now too. `SubscribeToLive` only completes when the subscription is confirmed by the subscription service, and the `MainLoop` is awaiting `SubscribeToLive`. so I think the MainLoop can just call `ConfirmSubscription` after `SubscribeToLive` completes.

I think that would remove the need for `OnSubscriptionMessage` and `OnMessage` to call `ConfirmSubscription`, and then since the call to ConfirmSubscription is only coming from 1 place and at a well known moment that ought to remove the need for the `_subscriptionStarted` variable and the compare-exchange

Reply:
great idea, thanks, i've made the changes accordingly